nntrainer/opencl/opencl_loader.cpp

```diff
@@ -65,6 +65,101 @@ bool LoadOpenCL() {
+    SWITCH_CASE_RETURN(CL_IMAGE_FORMAT_MISMATCH);
+    SWITCH_CASE_RETURN(CL_IMAGE_FORMAT_NOT_SUPPORTED);
+    SWITCH_CASE_RETURN(CL_BUILD_PROGRAM_FAILURE);
+    SWITCH_CASE_RETURN(CL_MAP_FAILURE);
+#ifdef CL_VERSION_1_1
+    SWITCH_CASE_RETURN(CL_MISALIGNED_SUB_BUFFER_OFFSET);
+    SWITCH_CASE_RETURN(CL_EXEC_STATUS_ERROR_FOR_EVENTS_IN_WAIT_LIST);
+#endif
+#ifdef CL_VERSION_1_2
+    SWITCH_CASE_RETURN(CL_COMPILE_PROGRAM_FAILURE);
+    SWITCH_CASE_RETURN(CL_LINKER_NOT_AVAILABLE);
+    SWITCH_CASE_RETURN(CL_LINK_PROGRAM_FAILURE);
+    SWITCH_CASE_RETURN(CL_DEVICE_PARTITION_FAILED);
+    SWITCH_CASE_RETURN(CL_KERNEL_ARG_INFO_NOT_AVAILABLE);
+#endif
+    SWITCH_CASE_RETURN(CL_INVALID_VALUE);
+    SWITCH_CASE_RETURN(CL_INVALID_DEVICE_TYPE);
+    SWITCH_CASE_RETURN(CL_INVALID_PLATFORM);
+    SWITCH_CASE_RETURN(CL_INVALID_DEVICE);
+    SWITCH_CASE_RETURN(CL_INVALID_CONTEXT);
+    SWITCH_CASE_RETURN(CL_INVALID_QUEUE_PROPERTIES);
+    SWITCH_CASE_RETURN(CL_INVALID_COMMAND_QUEUE);
+    SWITCH_CASE_RETURN(CL_INVALID_HOST_PTR);
+    SWITCH_CASE_RETURN(CL_INVALID_MEM_OBJECT);
+    SWITCH_CASE_RETURN(CL_INVALID_IMAGE_FORMAT_DESCRIPTOR);
+    SWITCH_CASE_RETURN(CL_INVALID_IMAGE_SIZE);
+    SWITCH_CASE_RETURN(CL_INVALID_SAMPLER);
+    SWITCH_CASE_RETURN(CL_INVALID_BINARY);
+    SWITCH_CASE_RETURN(CL_INVALID_BUILD_OPTIONS);
+    SWITCH_CASE_RETURN(CL_INVALID_PROGRAM);
+    SWITCH_CASE_RETURN(CL_INVALID_PROGRAM_EXECUTABLE);
+    SWITCH_CASE_RETURN(CL_INVALID_KERNEL_NAME);
+    SWITCH_CASE_RETURN(CL_INVALID_KERNEL_DEFINITION);
+    SWITCH_CASE_RETURN(CL_INVALID_KERNEL);
+    SWITCH_CASE_RETURN(CL_INVALID_ARG_INDEX);
+    SWITCH_CASE_RETURN(CL_INVALID_ARG_VALUE);
+    SWITCH_CASE_RETURN(CL_INVALID_ARG_SIZE);
+    SWITCH_CASE_RETURN(CL_INVALID_KERNEL_ARGS);
+    SWITCH_CASE_RETURN(CL_INVALID_WORK_DIMENSION);
+    SWITCH_CASE_RETURN(CL_INVALID_WORK_GROUP_SIZE);
+    SWITCH_CASE_RETURN(CL_INVALID_WORK_ITEM_SIZE);
+    SWITCH_CASE_RETURN(CL_INVALID_GLOBAL_OFFSET);
+    SWITCH_CASE_RETURN(CL_INVALID_EVENT_WAIT_LIST);
+    SWITCH_CASE_RETURN(CL_INVALID_EVENT);
+    SWITCH_CASE_RETURN(CL_INVALID_OPERATION);
+    SWITCH_CASE_RETURN(CL_INVALID_GL_OBJECT);
+    SWITCH_CASE_RETURN(CL_INVALID_BUFFER_SIZE);
+    SWITCH_CASE_RETURN(CL_INVALID_MIP_LEVEL);
+    SWITCH_CASE_RETURN(CL_INVALID_GLOBAL_WORK_SIZE);
+#ifdef CL_VERSION_1_1
+    SWITCH_CASE_RETURN(CL_INVALID_PROPERTY);
+#endif
+#ifdef CL_VERSION_1_2
+    SWITCH_CASE_RETURN(CL_INVALID_IMAGE_DESCRIPTOR);
+    SWITCH_CASE_RETURN(CL_INVALID_COMPILER_OPTIONS);
+    SWITCH_CASE_RETURN(CL_INVALID_LINKER_OPTIONS);
+    SWITCH_CASE_RETURN(CL_INVALID_DEVICE_PARTITION_COUNT);
+#endif
+#ifdef CL_VERSION_2_0
+    SWITCH_CASE_RETURN(CL_INVALID_PIPE_SIZE);
+    SWITCH_CASE_RETURN(CL_INVALID_DEVICE_QUEUE);
+#endif
+#ifdef CL_VERSION_2_2
+    SWITCH_CASE_RETURN(CL_INVALID_SPEC_ID);
+    SWITCH_CASE_RETURN(CL_MAX_SIZE_RESTRICTION_EXCEEDED);
+#endif
+  default:
+    return "(unknown)";
+  }
+
+  return "(unknown)";
```

Comment:
The return "(unknown)"; statement at the end of the function will never be executed, because it has already been returned from the default case of the switch statement.
Let's remove line 158.